realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -308,8 +308,28 @@ public void onCall() {
      * {@link io.realm.Realm#cancelTransaction()}. Transactions are used to atomically create, update and delete objects
      * within a Realm.
      * <p>
-     * Before beginning the transaction, {@link io.realm.Realm#beginTransaction()} updates the Realm in the case of
-     * pending updates from other threads.
+     * Writing to a Realm always happens on the latest version of the data. So before beginning a transaction, this
+     * instance of Realm is updated to the latest version that includes changes from all threads. This update does not
+     * trigger any registered {@link RealmChangeListener}.
+     * <p>
+     * For that reason, it is best practise to query for any data that should be modified from inside the
```

Comment:
It is therefore recommended to query for data from within the transaction that is going to modify it in order to avoid the risk of data mismatch.